astropy/stats/funcs.py

```diff
@@ -766,26 +778,59 @@ def median_absolute_deviation(a, axis=None, func=None):
         # See https://github.com/numpy/numpy/issues/7330 why using np.ma.median
         # for normal arrays should not be done (summary: np.ma.median always
         # returns an masked array even if the result should be scalar). (#4658)
-        if isinstance(a, np.ma.MaskedArray):
+        if isinstance(data, np.ma.MaskedArray):
+            is_masked = True
             func = np.ma.median
+            if ignore_nan:
+                data = np.ma.masked_where(np.isnan(data), data)
+        elif ignore_nan:
+            is_masked = False
+            if NUMPY_LT_1_10:
```

Comment:
I'm confused - help on `np.nanmedian` states this has been present since numpy 1.9, and I just checked that this is indeed the case (contrary to what was stated in https://github.com/astropy/astropy/pull/5232#issuecomment-239133691)..  Since we just dropped support for 1.7 and 1.8, it would seem this means we can get rid of this stanza!